### PR TITLE
Fix/night lag

### PR DESCRIPTION
**Pull request by Raycoms**
This PR should decrease the lag significantly!

Issue:

Each time on a server night approaches ALL colonies update their happiness and ALL mark themselves dirty and ALL have to write to file. (Depending on the server this is up to 500 colonies).

Solution:

We only update if a subscriber is around (Officer or similar).

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/colony/Colony.java [hunk 1]
```diff
@@ -709,11 +711,14 @@ private void updateWayPoints()
...
             {
                 @NotNull final BlockPos key = (BlockPos) ((Map.Entry) obj).getKey();
-                @NotNull final IBlockState value = (IBlockState) ((Map.Entry) obj).getValue();
-                if (world != null && world.getBlockState(key).getBlock() != (value.getBlock()))
+                if (world.isBlockLoaded(key))
                 {
-                    wayPoints.remove(key);
-                    markDirty();
+                    @NotNull final IBlockState value = (IBlockState) ((Map.Entry) obj).getValue();
+                    if (world != null && world.getBlockState(key).getBlock() != (value.getBlock()))
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/colony/Colony.java]: MAJOR Remove this expression which always evaluates to "true" [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2589)

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 52 issues
* ![MAJOR][MAJOR] 51 major
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/9850a54ab2017e2a88f9c8e04c1cfbe7e0744e3d/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/9850a54ab2017e2a88f9c8e04c1cfbe7e0744e3d/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [render#](https://github.com/Minecolonies/minecolonies/blob/9850a54ab2017e2a88f9c8e04c1cfbe7e0744e3d/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [Colony.java#L58](https://github.com/Minecolonies/minecolonies/blob/9850a54ab2017e2a88f9c8e04c1cfbe7e0744e3d/src/main/java/com/minecolonies/coremod/colony/Colony.java#L58): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 38 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L58](https://github.com/Minecolonies/minecolonies/blob/9850a54ab2017e2a88f9c8e04c1cfbe7e0744e3d/src/main/java/com/minecolonies/coremod/colony/Colony.java#L58): class "Colony" has 67 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [buildings#](https://github.com/Minecolonies/minecolonies/blob/9850a54ab2017e2a88f9c8e04c1cfbe7e0744e3d/src/main/java/com/minecolonies/coremod/colony/buildings#): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [permissions#](https://github.com/Minecolonies/minecolonies/blob/9850a54ab2017e2a88f9c8e04c1cfbe7e0744e3d/src/main/java/com/minecolonies/coremod/colony/permissions#): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [requestable#](https://github.com/Minecolonies/minecolonies/blob/9850a54ab2017e2a88f9c8e04c1cfbe7e0744e3d/src/main/java/com/minecolonies/coremod/colony/requestable#): Add a 'package-info.java' file to document the 'requestable' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [data#](https://github.com/Minecolonies/minecolonies/blob/9850a54ab2017e2a88f9c8e04c1cfbe7e0744e3d/src/main/java/com/minecolonies/coremod/colony/requestsystem/data#): Add a 'package-info.java' file to document the 'data' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [init#](https://github.com/Minecolonies/minecolonies/blob/9850a54ab2017e2a88f9c8e04c1cfbe7e0744e3d/src/main/java/com/minecolonies/coremod/colony/requestsystem/init#): Add a 'package-info.java' file to document the 'init' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 51 issues
* ![MAJOR][MAJOR] 50 major
* ![INFO][INFO] 1 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/64505fe8c923ef96945aa7cba922714b78e92e08/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/64505fe8c923ef96945aa7cba922714b78e92e08/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [render#](https://github.com/Minecolonies/minecolonies/blob/64505fe8c923ef96945aa7cba922714b78e92e08/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [Colony.java#L58](https://github.com/Minecolonies/minecolonies/blob/64505fe8c923ef96945aa7cba922714b78e92e08/src/main/java/com/minecolonies/coremod/colony/Colony.java#L58): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 37 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L58](https://github.com/Minecolonies/minecolonies/blob/64505fe8c923ef96945aa7cba922714b78e92e08/src/main/java/com/minecolonies/coremod/colony/Colony.java#L58): class "Colony" has 67 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [buildings#](https://github.com/Minecolonies/minecolonies/blob/64505fe8c923ef96945aa7cba922714b78e92e08/src/main/java/com/minecolonies/coremod/colony/buildings#): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [permissions#](https://github.com/Minecolonies/minecolonies/blob/64505fe8c923ef96945aa7cba922714b78e92e08/src/main/java/com/minecolonies/coremod/colony/permissions#): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [requestable#](https://github.com/Minecolonies/minecolonies/blob/64505fe8c923ef96945aa7cba922714b78e92e08/src/main/java/com/minecolonies/coremod/colony/requestable#): Add a 'package-info.java' file to document the 'requestable' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [data#](https://github.com/Minecolonies/minecolonies/blob/64505fe8c923ef96945aa7cba922714b78e92e08/src/main/java/com/minecolonies/coremod/colony/requestsystem/data#): Add a 'package-info.java' file to document the 'data' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [init#](https://github.com/Minecolonies/minecolonies/blob/64505fe8c923ef96945aa7cba922714b78e92e08/src/main/java/com/minecolonies/coremod/colony/requestsystem/init#): Add a 'package-info.java' file to document the 'init' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by Asherslab** [comment]
Looks good to me :lgmt:

**Review by Asherslab — APPROVED** [review]

**Comment by marchermans** [comment]
We should really look into changing our data structures to be more effective. Having to do io everything is hurting us left and right.

**Review by marchermans — APPROVED** [review]
See normal comment. But approved.

**Pull request merged**